shuup/admin/modules/products/views/copy.py

```diff
@@ -0,0 +1,75 @@
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import DetailView
+
+from shuup.admin.utils.urls import get_model_url
+from shuup.core.models import (
+    Product, ProductAttribute, ProductMedia, ShopProduct
+)
+from shuup.customer_group_pricing.models import CgpPrice
+from shuup.simple_supplier.models import StockCount
+from shuup.utils.models import get_data_dict
+
+
+class ProductCopyView(DetailView):
+    model = ShopProduct
+    context_object_name = "product"
+
+    @atomic()
+    def get(self, request, *args, **kwargs):
+        shop_product = self.get_object()
+        product = shop_product.product
+        product_data = get_data_dict(product)
+        product_data.update({
+            "sku": "{}-{}".format(product.sku, Product.objects.count())
+        })
+        new_product = Product.objects.create(**product_data)
+        new_product.name = product.name
+        new_product.short_description = product.short_description
+        new_product.description = product.description
+        new_product.slug = product.slug
+        new_product.save()
+
+        new_shop_product = ShopProduct.objects.create(product=new_product, shop=shop_product.shop)
+        new_shop_product.visibility = shop_product.visibility
+        new_shop_product.purchasable = shop_product.purchasable
+        new_shop_product.default_price_value = shop_product.default_price_value
+        new_shop_product.primary_category = shop_product.primary_category
+        new_shop_product.save()
+
+        new_shop_product.suppliers = shop_product.suppliers.all()
+        new_shop_product.categories = shop_product.categories.all()
+
+        for attribute in product.attributes.all():
+            ProductAttribute.objects.create(product=new_product, attribute=attribute.attribute, value=attribute.value)
+
+        for old_media in product.media.all():
+            media = ProductMedia.objects.create(product=new_product, file=old_media.file, kind=old_media.kind)
+            media.shops.add(shop_product.shop)
+            if product.primary_image == old_media:
+                new_product.primary_image = media
+                new_product.save()
+
+        for price in CgpPrice.objects.filter(product=product, shop=shop_product.shop):
+            CgpPrice.objects.create(
+                product=new_product, shop=shop_product.shop, group=price.group,
+                price_value=price.price_value
+            )
+
+        for supplier in shop_product.suppliers.all():
+            origin_product_stock_count = StockCount.objects.get_or_create(supplier=supplier, product=product)[0]
+            new_product_stock_count = StockCount.objects.get_or_create(supplier=supplier, product=new_product)[0]
+            new_product_stock_count.stock_managed = origin_product_stock_count.stock_managed
+            new_product_stock_count.save(update_fields=["stock_managed"])
+
+        messages.success(request, _("Product %s copy successfull.") % new_product)
```

Comment:
I would recommend you to trigger some custom signal here like:
```py
product_copied.send(sender=type(self), copied=product, copy=new_prouct)
```

this way, other addons can do their copy part if they need to (this is the case of the supplier module and others).